### PR TITLE
feat(HACBS-2526): update tekton api to v1

### PR DESCRIPTION
**Pull request by Troy876**
Updates the tasks and pipelines to Tekton API v1.

### CODE REVIEW

**Comment by johnbieren** [comment]
> @Troy876 can you provide diffs of what was actually changed? Either you can paste it as a comment, but it would be preferable to have a separate commit for cloning the pipelines/tasks and then for the actual change (see [here](https://github.com/redhat-appstudio/release-service-bundles/pull/191) for an example).
> 
> Also, do we even need a new version for this? @davidmogar @johnbieren thoughs?

I think a new version is good here. I haven't done any testing of pipeline definitions going from `v1beta1` -> `v1` so not sure what the effects are, but if there are any, I don't want e2e to break

**Comment by johnbieren** [comment]
@kasemAlem can you work with Troy on this to show him running the e2e-tests with these new pipeline versions? I don't know if this change will break e2e or not

**Comment by kasemAlem** [comment]
> @kasemAlem can you work with Troy on this to show him running the e2e-tests with these new pipeline versions? I don't know if this change will break e2e or not

Sure I can do that ,

**Comment by happybhati** [comment]
@Troy876 The Tekton Task Test is now fixed. You need to rebase your PR with the latest release-service-bundle, and you are good to go.

**Review by johnbieren — COMMENTED** [review]
The resolvers in the pipelines were not updated to use the new tasks so with this change none of the tasks added here are in use

**Comment by Troy876** [comment]
> The resolvers in the pipelines were not updated to use the new tasks so with this change none of the tasks added here are in use

I've updated the resolvers to use the new versions, should be fine now

**Comment by mmalina** [comment]
> > The resolvers in the pipelines were not updated to use the new tasks so with this change none of the tasks added here are in use
> 
> I've updated the resolvers to use the new versions, should be fine now

Can you provide the diffs as I asked [here](https://github.com/redhat-appstudio/release-service-bundles/pull/192#issuecomment-1713761559)? Of course I can do it myself, but it's much more efficient to do it once compared to every reviewer doing it on their own.

**Comment by Troy876** [comment]
> > > The resolvers in the pipelines were not updated to use the new tasks so with this change none of the tasks added here are in use
> > 
> > 
> > I've updated the resolvers to use the new versions, should be fine now
> 
> Can you provide the diffs as I asked [here](https://github.com/redhat-appstudio/release-service-bundles/pull/192#issuecomment-1713761559)? Of course I can do it myself, but it's much more efficient to do it once compared to every reviewer doing it on their own.

Sure, I've just no idea how, mind telling me?

File: catalog/task/update-ocp-tag/0.2/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+## Parameters
+
+| Name | Description | Optional | Default value |
+|------|-------------|----------|---------------|
+| fromIndex | The source Index image (catalog of catalogs) FBC fragment | No | - |
+| targetIndex | Index image (catalog of catalogs) the FBC fragment will be added to | No | - |
+| binaryImage | OCP binary image to be baked into the index image | No | - |
+| ocpVersion |  OCP version fetched from fbcFragment | No | - |
+
+## Chnages since 0.1
```
Comment (johnbieren) [inline catalog/task/update-ocp-tag/0.2/README.md:16]: typo. Changes not Chnages

**Review by johnbieren — COMMENTED** [review]
Besides the typo, LGTM. Nice job!

**Comment by mmalina** [comment]
@Troy876 note that @johnbieren added version 0.21 of the fbc-release pipeline here: https://github.com/redhat-appstudio/release-service-bundles/pull/203
So you will have to rebase and add 0.22 instead.

**Comment by mmalina** [comment]
> > Can you provide the diffs as I asked [here](https://github.com/redhat-appstudio/release-service-bundles/pull/192#issuecomment-1713761559)? Of course I can do it myself, but it's much more efficient to do it once compared to every reviewer doing it on their own.
> 
> Sure, I've just no idea how, mind telling me?

It's best to provide two commits like Johnny did here for example: https://github.com/redhat-appstudio/release-service-bundles/pull/203/commits - the first commit just adds the new versions (duplicating the old versions) and the second commit makes the changes to those new versions. That way it's easy to review the actual changes. Note that once we move away from this version structure (hopefully soon), this will no longer be a problem.

Another option is to provide diffs of the new version as a comment. So for each new version that you add, you will run the `diff` command and post the result here. For example, say you're adding version 0.6 of the create-internal-request task. So you would run diff between the previous and the new version: `diff -r catalog/task/create-internal-request/0.5 catalog/task/create-internal-request/0.6` and post the result. You would do this for all the tasks and pipelines that you add a new version for.

I guess for now it's not needed anymore as Johnny reviewed your changes (he likely checked out your branch and ran diff manually on his computer) and you're good to go. But it's something to keep in mind for next time (until we migrate the repo to the new layout).

**Comment by johnbieren** [comment]
> > > Can you provide the diffs as I asked [here](https://github.com/redhat-appstudio/release-service-bundles/pull/192#issuecomment-1713761559)? Of course I can do it myself, but it's much more efficient to do it once compared to every reviewer doing it on their own.
> > 
> > 
> > Sure, I've just no idea how, mind telling me?
> 
> It's best to provide two commits like Johnny did here for example: https://github.com/redhat-appstudio/release-service-bundles/pull/203/commits - the first commit just adds the new versions (duplicating the old versions) and the second commit makes the changes to those new versions. That way it's easy to review the actual changes. Note that once we move away from this version structure (hopefully soon), this will no longer be a problem.
> 
> Another option is to provide diffs of the new version as a comment. So for each new version that you add, you will run the `diff` command and post the result here. For example, say you're adding version 0.6 of the create-internal-request task. So you would run diff between the previous and the new version: `diff -r catalog/task/create-internal-request/0.5 catalog/task/create-internal-request/0.6` and post the result. You would do this for all the tasks and pipelines that you add a new version for.
> 
> I guess for now it's not needed anymore as Johnny reviewed your changes (he likely checked out your branch and ran diff manually on his computer) and you're good to go. But it's something to keep in mind for next time (until we migrate the repo to the new layout).

Yeah, I cloned it and reviewed it pretty closely. Hoping to get this merged ASAP since it is so large

File: catalog/task/extract-index-image/0.2/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# publish-index-image
```
Comment (johnbieren) [inline catalog/task/extract-index-image/0.2/README.md:1]: @Troy876 this is my fault, not yours, but can you change this to `extract-index-image`?

File: catalog/pipeline/fbc-release/0.22/README.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+| fbcPublishingCredentials | Secret used to publish the built index image | Yes | "fbc-publishing-credentials" |
+| requestUpdateTimeout | Max seconds to wait until the status is updated | Yes | - |
+| buildTimeoutSeconds | Max seconds to wait until the build finishes | Yes | - |
+| verify_ec_task_git_url | The git repo url of the verify-enterprise-contract task | No | - |
+| verify_ec_task_git_revision | The git repo revision the verify-enterprise-contract task | No | - |
+| verify_ec_task_git_pathInRepo | The location of the verify-enterprise-contract task in its repo | No | - |
+
+## Changelog
+
+### Changes since 0.21
```
Comment (johnbieren) [inline catalog/pipeline/fbc-release/0.22/README.md:33]: Forgot to list the change after this line

**Review by johnbieren — COMMENTED** [review]
Two small fixes

**Comment by theflockers** [comment]
I created a gist with the latest diffs https://gist.github.com/theflockers/b2e57c68be1ec7ecd5c42a46305547fb

File: catalog/pipeline/push-to-external-registry/0.19/README.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+| addTimestampTag | When pushing the snapshot components, also push a tag with the current timestamp | Yes | false |
+| pyxisServerType | The Pyxis server type to use. Options are 'production' and 'stage' | No | - |
+| pyxisSecret | The kubernetes secret to use to authenticate to Pyxis. It needs to contain two keys: key and cert | No | - |
+| postCleanUp | Cleans up workspace after finishing executing the pipeline | Yes | true |
+| verify_ec_task_git_url | The git repo url of the verify-enterprise-contract task | No | - |
+| verify_ec_task_git_revision | The git repo revision the verify-enterprise-contract task | No | - |
+| verify_ec_task_git_pathInRepo | The location of the verify-enterprise-contract task in its repo | No | - |
+
+## Changes since 0.18
+- update Tekton API to v1
```
Comment (theflockers) [inline catalog/pipeline/push-to-external-registry/0.19/README.md:31]: Same as my previous comment about the task versions. The same comment is valid for all pipelines README

File: catalog/task/create-internal-request/0.8/README.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+| request | Internal pipeline request name | No | |
+| params | Internal Request parameters | No | |
+| inputDataFile | Optional file to read data from | Yes | "" |
+| updateGenericResult | Should the task update the genericResult result  | Yes | "false" |
+| requestUpdateTimeout | Max seconds to wait until the status is updated | Yes | 360 |
+
+## Changelog
+
+### changes since 0.7
+- ipdate Tekton API to v1
```
Comment (theflockers) [inline catalog/task/create-internal-request/0.8/README.md:19]:
```suggestion
- Update Tekton API to v1
```

**Review by theflockers — CHANGES_REQUESTED** [review]
small changes.

File: catalog/task/extract-index-image/0.2/README.md (hunk 2 above)
Comment (Troy876) [reply catalog/task/extract-index-image/0.2/README.md]: Should all good now

File: catalog/pipeline/fbc-release/0.22/README.md (hunk 3 above)
Comment (Troy876) [reply catalog/pipeline/fbc-release/0.22/README.md:33]: Listed the change and the tasks were bumped, should be good

File: catalog/pipeline/push-to-external-registry/0.19/README.md (hunk 4 above)
Comment (Troy876) [reply catalog/pipeline/push-to-external-registry/0.19/README.md:31]: Should be adjusted

File: catalog/task/create-internal-request/0.8/README.md (hunk 5 above)
Comment (Troy876) [reply catalog/task/create-internal-request/0.8/README.md]: Fixed typo

**Review by johnbieren — DISMISSED** [review]

**Review by theflockers — DISMISSED** [review]

**Review by happybhati — DISMISSED** [review]

File: catalog/pipeline/fbc-release/0.22/fbc-release.yaml [hunk 6]
```diff
@@ -0,0 +1,374 @@
...
+    - name: postCleanUp
+      type: string
+      description: Cleans up workspace after finishing executing the pipeline
+      default: "true"
+  workspaces:
+    - name: release-workspace
+  results:
+    - name: requestMessage
+      value: $(tasks.add-fbc-contribution-to-index-image.results.requestMessage)
+    - name: requestReason
```
Comment (johnbieren) [inline catalog/pipeline/fbc-release/0.22/fbc-release.yaml:96]: Can you add `type: string` to this result?

File: catalog/pipeline/fbc-release/0.22/fbc-release.yaml [hunk 7]
```diff
@@ -0,0 +1,374 @@
...
+      value: $(tasks.add-fbc-contribution-to-index-image.results.requestReason)
+    - name: iibIndexImage
```
Comment (johnbieren) [inline catalog/pipeline/fbc-release/0.22/fbc-release.yaml:98]: Can you add `type: string` to this result?

File: catalog/pipeline/fbc-release/0.22/fbc-release.yaml [hunk 8]
```diff
@@ -0,0 +1,374 @@
...
+      value: $(tasks.extract-index-image.results.indexImage)
+    - name: iibIndexImageResolved
```
Comment (johnbieren) [inline catalog/pipeline/fbc-release/0.22/fbc-release.yaml:100]: Can you add `type: string` to this result?

File: catalog/task/apply-mapping/0.7/apply-mapping.yaml [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+      description: The path to the config file containing the mapping
+      default: ""
+    - name: failOnEmptyResult
+      type: string
+      description: Fail the task if the resulting snapshot contains 0 components
+      default: "false"
+  workspaces:
+    - name: config
+      description: The workspace where the extra config file containing the mapping and snapshot json reside
+  results:
```
Comment (johnbieren) [inline catalog/task/apply-mapping/0.7/apply-mapping.yaml:30]: Can you add `type: string` to this result?

File: catalog/task/collect-data/0.4/collect-data.yaml [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+      description: The namespaced name of the Snapshot
+    - name: subdirectory
+      description: Subdirectory inside the workspace to be used
+      type: string
+      default: ""
+  workspaces:
+    - name: data
+      description: Workspace to save the CR jsons to
+  results:
+    - name: release
```
Comment (johnbieren) [inline catalog/task/collect-data/0.4/collect-data.yaml:38]: Can you add `type: string` to this result?

File: catalog/task/collect-data/0.4/collect-data.yaml [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+      description: The relative path in the workspace to the stored release json
+    - name: releasePlan
```
Comment (johnbieren) [inline catalog/task/collect-data/0.4/collect-data.yaml:40]: Can you add `type: string` to this result?

File: catalog/task/collect-data/0.4/collect-data.yaml [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+      description: The relative path in the workspace to the stored releasePlan json
+    - name: releasePlanAdmission
```
Comment (johnbieren) [inline catalog/task/collect-data/0.4/collect-data.yaml:42]: Can you add `type: string` to this result?

File: catalog/task/collect-data/0.4/collect-data.yaml [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+      description: The relative path in the workspace to the stored releasePlanAdmission json
+    - name: releaseStrategy
```
Comment (johnbieren) [inline catalog/task/collect-data/0.4/collect-data.yaml:44]: Can you add `type: string` to this result?

File: catalog/task/collect-data/0.4/collect-data.yaml [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+      description: The relative path in the workspace to the stored releaseStrategy json
+    - name: snapshotSpec
+      description: The relative path in the workspace to the stored snapshotSpec json
+    - name: fbcFragment
```
Comment (johnbieren) [inline catalog/task/collect-data/0.4/collect-data.yaml:48]: Can you add `type: string` to this result?

File: catalog/task/collect-data/0.4/collect-data.yaml [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+      description: The first component's containerImage in the snapshot to use in the fbc pipelines
+    - name: extraData
```
Comment (johnbieren) [inline catalog/task/collect-data/0.4/collect-data.yaml:50]: Can you add `type: string` to this result?

File: catalog/task/create-internal-request/0.8/create-internal-request.yaml [hunk 16]
```diff
@@ -0,0 +1,187 @@
...
+    - name: subdirectory
+      description: Subdirectory inside the workspace to be used
+      type: string
+      default: ""
+  results:
+    - name: requestMessage
+      description: Internal Request message
+    - name: requestReason
+      description: Internal Request reason
+    - name: requestResultsFile
```
Comment (johnbieren) [inline catalog/task/create-internal-request/0.8/create-internal-request.yaml:44]: Can you add `type: string` to this result?

File: catalog/task/extract-index-image/0.2/extract-index-image.yaml [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+  description: >-
+      Extract the index image fields from the inputDataFile
+  params:
+    - name: inputDataFile
+      type: string
+  workspaces:
+    - name: input
+      description: Workspace where the inputDataFile is stored
+  results:
+    - name: indexImage
```
Comment (johnbieren) [inline catalog/task/extract-index-image/0.2/extract-index-image.yaml:21]: Can you add `type: string` to this result?

File: catalog/task/extract-index-image/0.2/extract-index-image.yaml [hunk 18]
```diff
@@ -0,0 +1,37 @@
...
+    - name: indexImageResolved
```
Comment (johnbieren) [inline catalog/task/extract-index-image/0.2/extract-index-image.yaml:22]: Can you add `type: string` to this result?

File: catalog/task/prepare-validation/0.4/prepare-validation.yaml [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task to extract a pull spec from a Snapshot
+  params:
+    - name: snapshot
+      type: string
+      description: The Snapshot in JSON format to apply the mapping to
+  results:
+    - name: snapshot
```
Comment (johnbieren) [inline catalog/task/prepare-validation/0.4/prepare-validation.yaml:19]: Can you add `type: string` to this result?

File: catalog/task/publish-index-image/0.3/publish-index-image.yaml [hunk 20]
```diff
@@ -0,0 +1,72 @@
...
+      description: The credentials used to access the registries
+    - name: requestUpdateTimeout
+      type: string
+      default: "360"
+      description: Max seconds waiting for the status update
+  workspaces:
+    - name: input
+      description: Workspace to store the params and responses for the internalRequest
+  results:
+    - name: requestMessage
```
Comment (johnbieren) [inline catalog/task/publish-index-image/0.3/publish-index-image.yaml:37]: Can you add `type: string` to this result?

File: catalog/task/sign-index-image/0.2/sign-index-image.yaml [hunk 21]
```diff
@@ -0,0 +1,117 @@
...
+    - name: configMapName
+      type: string
+      description: ConfigMap Name for this request
+    - name: pipelineRunName
+      type: string
+      description: The name of the Parent PipelineRun of this task
+  results:
+    - name: requestMessage
+      description: Internal Request message
+    - name: requestReason
```
Comment (johnbieren) [inline catalog/task/sign-index-image/0.2/sign-index-image.yaml:43]: Can you add `type: string` to this result?

File: catalog/task/update-ocp-tag/0.2/update-ocp-tag.yaml [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+    - name: binaryImage
+      description: OCP binary image to be baked into the index image
+      type: string
+    - name: ocpVersion
+      description: OCP version tag to replace the current set tags on index images
+      type: string
+  results:
+    - name: updated-fromIndex
+      description: source Index image (catalog of catalogs) FBC fragment with updated tag
+    - name: updated-targetIndex
```
Comment (johnbieren) [inline catalog/task/update-ocp-tag/0.2/update-ocp-tag.yaml:31]: Can you add `type: string` to this result?

File: catalog/task/update-ocp-tag/0.2/update-ocp-tag.yaml [hunk 23]
```diff
@@ -0,0 +1,59 @@
...
+      description: Index image (catalog of catalogs) the FBC fragment will be added to with updated tag
+    - name: updated-binaryImage
```
Comment (johnbieren) [inline catalog/task/update-ocp-tag/0.2/update-ocp-tag.yaml:33]: Can you add `type: string` to this result?

File: catalog/task/upstream/tekton/kubernetes-actions/0.3/kubernetes-actions.yaml [hunk 24]
```diff
@@ -0,0 +1,56 @@
...
+  description: >-
+    This task is the generic kubectl CLI task which can be used
+    to run all kinds of k8s commands
+  workspaces:
+    - name: manifest-dir
+      optional: true
+    - name: kubeconfig-dir
+      optional: true
+  results:
+    - name: output-result
```
Comment (johnbieren) [inline catalog/task/upstream/tekton/kubernetes-actions/0.3/kubernetes-actions.yaml:24]: Can you add `type: string` to this result?

File: catalog/pipeline/fbc-release/0.22/fbc-release.yaml [hunk 25]
```diff
@@ -0,0 +1,374 @@
...
+      type: string
+      description: The location of the verify-enterprise-contract task in its repo
+    - name: postCleanUp
+      type: string
+      description: Cleans up workspace after finishing executing the pipeline
+      default: "true"
+  workspaces:
+    - name: release-workspace
+  results:
+    - name: requestMessage
```
[the remark this answers is not in the document]
Comment (johnbieren) [reply catalog/pipeline/fbc-release/0.22/fbc-release.yaml:94]: You have stirng, not string here

[the remark this answers is not in the document]
Comment (Troy876) [reply catalog/pipeline/fbc-release/0.22/fbc-release.yaml:94]: Typos will be the death of me

File: catalog/task/create-pyxis-image/0.5/README.md [hunk 26]
```diff
@@ -0,0 +1,34 @@
...
+
+| Name | Description | Optional | Default value |
+|------|-------------|----------|---------------|
+| server | The server type to use. Options are 'production' and 'stage' | Yes | production |
+| pyxisSecret | The kubernetes secret to use to authenticate to Pyxis. It needs to contain two keys: key and cert | No | - |
+| certified | If set to true, the images will be marked as certified in their Pyxis entries | Yes | false |
+| tag | The tag to use when pushing the container image metadata to Pyxis | No | - |
+| isLatest | If set to true, the images will have a latest tag added with their Pyxis entries | Yes | false |
+| snapshotPath | Path to the JSON string of the mapped Snapshot spec in the data workspace | Yes | mapped_snapshot.json |
+
```
Comment (johnbieren) [inline catalog/task/create-pyxis-image/0.5/README.md:20]: Missing Changes since 0.4

Comment (Troy876) [reply catalog/task/create-pyxis-image/0.5/README.md:20]: Should be good now

**Review by johnbieren — APPROVED** [review]
Kasem is going to rerun the tests one more time then we should merge this

**Review by happybhati — APPROVED** [review]

**Review by kasemAlem — APPROVED** [review]

**Pull request merged**